snapcraft/plugins/go.py

```diff
@@ -101,7 +102,13 @@ def get_pull_properties(cls):
 
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
-        self.build_packages.append('golang-go')
+
+        # Do not try to install golang if it is already available in our PATH.
+        try:
+            subprocess.check_output(['go', 'version'])
+        except OSError:
+            self.build_packages.append('golang-go')
```

Comment:
Could the go plugin not do what the Rust plugin does and allow the user to configure a specific version of the compiler to use?